src/server/dflycmd.cc

```diff
@@ -461,8 +461,8 @@ void DflyCmd::TakeOver(CmdArgList args, ConnectionContext* cntx) {
     shard_set->RunBlockingInParallel(std::move(cb));
   }
 
+  sf_->service().SwitchState(GlobalState::TAKEN_OVER, GlobalState::ACTIVE);
```

Comment:
why?

Reply:
we have the  save operation before shutdown. With the new rules we can not do save during takeOver and the state should be active if we want to do the save. We already finished the takeOver operation so we must change the state.

Reply:
I believe we should not change the state here back to ACTIVE the server should stay in TAKEN_OVER state.
This state is protecting quering the server issuing new commands untill the shutdow is finished, and this is also the actual state of the server until it finishes the shutdown.

If we want to allow the save operation after the shutdown that we should allow switching from ACTIVE or from TAKEN_OVER in the save operation

Reply:
we already do the save operation before shutdown and if we can not change the state we shouldn't do the save.
Also like an option we can create an additional state or modify our save function for shutdown purposes, I will check what will be better

Reply:
We are calling the save operation in the shutdown call below. Before you PR changes the state here was TAKEN_OVER and than when calling shutdown the save would be rejected becuase we are not in active correct?
why are you trying to change this right now so we will save on shutdown after TAKEOVER?

Reply:
we had a discussion in the Eng chat that we shouldn't do save during TAKEOVER. So I want to provide some guarantees for it.